### PR TITLE
Bugfix: fix audio popping clicking

### PR DESCRIPTION
**Pull request by gozmanyoni**
This PR Fixes audio playback in the examples.

The reasons audio was choppy/noisy before is:
1. We used to feed Buffer.buffer to the PCM player - .buffer point to the original arrayBuffer on data we get from the driver, and it always starts with 3 integers that indicate audio type, volume and command - this is treated as audio and played - and causes pops and clicks
2. The pcm-player used is heavy on intervals and object creation, causing a lot of GC and eating away at the main threads capacity - pcm-ringbuf-player uses shared memory between the main and audio thread, and leverages a ring-buffer data structure to efficiently push and read audio with no locks or array creation.

Also - The driver now emits typed Singed In16 data (instead of a buffer), it will create the array at the correct offset, making sure that in can be safely consumed later without accidentally including the control data

### CODE REVIEW

File: examples/carplay-web-app-direct/src/setupProxy.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = function (app) {
```
Comment (gozmanyoni) [inline examples/carplay-web-app-direct/src/setupProxy.js:1]: Needed for SharedArrayBuffers

File: src/modules/messages.ts [hunk 2]
```diff
@@ -325,16 +325,15 @@ export const decodeTypeMap: DecodeTypeMapping = {
 
 export class AudioData extends Message {
   command?: AudioCommand
-  format: AudioFormat
+  decodeType: number
```
Comment (gozmanyoni) [inline src/modules/messages.ts:328]: Resolving this on the FE using mapping instead - allows us to use Format objects to reference AudioPlayers in the worker example, and minimises closed data over worker messages

**Comment by steelbrain** [comment]
Test methodology: Listening to "Snow on the beach" by Taylor Swift and paying attention to ~9 second mark.

In direct mode, still some audio artifacts, here's the console log



In worker mode, almost perfect, nothing noticable unless you're paying very close attention. Here's the log



I would say the worker one is good enough as the APK. I am testing this on an m1 macbook, will test on a raspberry pi with its capabilities to see what the results are

**Comment by gozmanyoni** [comment]
Overall - I think there is no reason to use it in direct mode unless one has a capabale machine. On a pi it should be the default

Its ideal to have the usb reading on a separate thread than the UI thread. the audio thread is separate by default - the apk also has seperate thread for audio,usb and video

**Comment by steelbrain** [comment]
Excited how all of this is coming together nicely! 🤩

**Pull request merged**